### DISCUSSION TITLE
How to set integrated two-step simulation

### DISCUSSION THREAD

**Discussion by zzzmx-josh** [idaholab/moose#27596] [Q&A General] [answered]
Hi there!

I'd like to conduct a two-step simulation:

Step 1: Apply a displacement in the z-direction to the mesh on the right. Use a steady-state simulation to obtain the deformed mesh and related displacement variables.

Step 2: Read the displacements from the deformed mesh for subsequent calculations. At this point, displacements are coupled with other variables, and the calculations are transient.

How can I easily achieve this simulation? Is it possible to do it in a single input file?

Thanks in advance!

**Comment by lindsayad** [comment, marked as answer]
This kind of thing is commonly done with MOOSE. The steady and transient simulations will require two different inputs. I would recommend reading [this page](https://mooseframework.inl.gov/application_usage/restart_recover.html) for a background on restart capabilities in MOOSE

**Reply by zzzmx-josh** [reply]
Thank you for the recommendation!
